app/helpers/sessions_helper.rb

```diff
@@ -3,15 +3,37 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
```

Comment:
cookies là gì em, hoạt động của nó như thế nào, và nó lưu trữ dữ liệu như thế nào?

Reply:
> cookies là gì em, hoạt động của nó như thế nào, và nó lưu trữ dữ liệu như thế nào?

cookies là đoạn kí tự nó lưu trên máy của mình. hoạt động của nó là: khi ví dụ mình log in vào trang framgia vs tài khoản ABC thì nó lưu cookies là 1 đoạn mã trên trình duyệt của mình, khi mình vào lại trang web framgia thì nó sẽ gửi đoạn mã kí tự đó lên server. server sẽ gửi lại những gì của tài khoản đó.